app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/AllPhotosAdapter.kt

```diff
@@ -49,9 +54,30 @@ class AllPhotosAdapter(): RecyclerView.Adapter<AllPhotosAdapter.ViewHolder>() {
                         holder.description.text = "fake description"
 
                         holder.image.setOnClickListener {
-                            val intent = Intent(context, ExpandPhotoActivity::class.java)
-                            intent.putExtra(ExpandPhotoActivity.EXTRA_INTENT_IMAGE_URI, picNote.imageUri)  // pass in 1 imageuri to ExpandPhotoActivity
-                            context.startActivity(intent)
+                            if(picNotesSelected.isNotEmpty()){
+                                if(picNotesSelected.contains(picNote)){
+                                    onOnImageActionListener?.onImageDeselected(picNote)
+                                } else {
+                                    onOnImageActionListener?.onImageSelected(picNote)
+                                }
+                            } else {
+                                val intent = Intent(context, ExpandPhotoActivity::class.java)
+                                intent.putExtra(ExpandPhotoActivity.EXTRA_INTENT_IMAGE_URI, picNote.imageUri)  // pass in 1 imageuri to ExpandPhotoActivity
+                                context.startActivity(intent)
+                            }
+                        }
+
+                        holder.image.setOnLongClickListener {
+                            onOnImageActionListener?.onImageSelected(picNote)
+                            true
+                        }
+
+                        if(picNotesSelected.contains(picNote)) {
+                            Log.i("INFO", "Marking " + picNote + " as selected!")
+                            val color = 0x8f000000.toInt()
```

Comment:
please put 0x8f000000.toInt() in a constant and give it a name, since we may want to reuse it

Reply:
kk